code/datums/preferences/preferences.dm

```diff
@@ -558,43 +558,43 @@ var/list/removed_jobs = list(
 			if ("update-flavorText")
 				var/new_text = tgui_input_text(usr, "Please enter new flavor text (appears when examining you):", "Character Generation", html_decode(src.flavor_text), multiline = TRUE, allowEmpty=TRUE)
 				if (!isnull(new_text))
-					new_text = html_encode(new_text)
 					if (length(new_text) > FLAVOR_CHAR_LIMIT)
 						tgui_alert(usr, "Your flavor text is too long. It must be no more than [FLAVOR_CHAR_LIMIT] characters long. The current text will be trimmed down to meet the limit.", "Flavor text too long")
 						new_text = copytext(new_text, 1, FLAVOR_CHAR_LIMIT+1)
+					new_text = html_encode(new_text)
 					src.flavor_text = new_text || null
 					src.profile_modified = TRUE
 					return TRUE
 
 			if ("update-securityNote")
 				var/new_text = tgui_input_text(usr, "Please enter new flavor text (appears when examining your security record):", "Character Generation", html_decode(src.security_note), multiline = TRUE, allowEmpty=TRUE)
 				if (!isnull(new_text))
-					new_text = html_encode(new_text)
 					if (length(new_text) > FLAVOR_CHAR_LIMIT)
-						tgui_alert(usr, "Your flavor text is too long. It must be no more than [FLAVOR_CHAR_LIMIT] characters long. The current text will be trimmed down to meet the limit.", "Flavor text too long")
+						tgui_alert(usr, "Your flavor text is too long. It must be no more than [MED_FLAVOR_CHAR_LIMIT] characters long. The current text will be trimmed down to meet the limit.", "Flavor text too long")
```

Comment:
You removed the `MED_FLAVOR_CHAR_LIMIT` define from an earlier commit, probably want to revert this line.

Reply:
Yeah oops, I should've just reverted that commit instead of this mess, fixed at least.